a10_octavia/tests/unit/controller/worker/tasks/test_service_group_tasks.py

```diff
@@ -36,9 +41,32 @@ def setUp(self):
         super(TestHandlerServiceGroupTasks, self).setUp()
         imp.reload(task)
         self.client_mock = mock.Mock()
+        self.conf = self.useFixture(oslo_fixture.Config(cfg.CONF))
+        self.conf.register_opts(A10_SERVICE_GROUP_OPTS,
+                                group=a10constants.SERVICE_GROUP_CONF_SECTION)
+
+    def tearDown(self):
+        super(TestHandlerServiceGroupTasks, self).tearDown()
+        self.conf.reset()
 
     def test_revert_pool_create_task(self):
         mock_pool = task.PoolCreate()
         mock_pool.axapi_client = self.client_mock
         mock_pool.revert(POOL, VTHUNDER)
         self.client_mock.slb.service_group.delete.assert_called_with(POOL.id)
+
+    def test_create_lb_algorithm_source_ip_hash_only(self):
+        mock_pool = task.PoolCreate()
+        mock_pool.axapi_client = self.client_mock
+        mock_pool.CONF = self.conf
+        POOL = o_data_models.Pool(id=a10constants.MOCK_POOL_ID,
+                                  protocol=a10constants.HTTP_PROTOCOL,
+                                  lb_algorithm=a10constants.POOL_LB_ALGO_SOURCE_IP)
+        mock_pool.execute(POOL, VTHUNDER)
+        self.client_mock.slb.service_group.create.assert_called_with(
```

Comment:
Since we only care about the lb_method in this test we can instead test with:
```python
self.client_mock.slb.service_group.create.assert_called_with(
    protocol=mock.ANY,
    lb_method=mock_pool.axapi_client.slb.service_group.SOURCE_IP_HASH_ONLY,
    service_group_templates=mock.ANY,
    axapi_args=AXAPI_ARGS)